docs/HowToFAQ/FAQRecursiveTermination.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Is there a simple way to prove the termination of a recursive function?
+---
+
+## Question
+
+Is there a simple way to prove the termination of a recursive function?
+
+Here is an example function:
+```dafny
+datatype Dummy = State1 | State2
+
+  function WalkState(str: string, s: Dummy): string {
+    if str == [] then []
+    else if s.State1? then WalkState(str[1..], Dummy.State2)
+    else WalkState(str, Dummy.State1)
+  }
+```
+
+## Answer
+
+In general, to prove termination of any recursive structure one needs to declare a 
+(well-founded) measure that decreases on each iteration or recursive invocation;
```

Comment:
Can you link to what a well-founded measure is, or at least describe it yourself? E.g. "A well-founded measure is an element in a set that has no infinitely decreasing sequences"

Reply:
Changed the wording and added a link